server/game/cards/02_SHD/units/TarffulKashyyykChieftain.ts

```diff
@@ -0,0 +1,36 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { DamageType, RelativePlayer, Trait, WildcardCardType, ZoneName } from '../../../core/Constants';
+
+export default class TarffulKashyyykChieftain extends NonLeaderUnitCard {
+    protected override getImplementationId () {
+        return {
+            id: '7022736145',
+            internalName: 'tarfful#kashyyyk-chieftain'
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.addTriggeredAbility({
+            title: 'That unit deals that much damage to an enemy ground unit',
+            when: {
+                onDamageDealt: (event, context) =>
+                    event.type === DamageType.Combat &&
+                    event.card.hasSomeTrait(Trait.Wookiee) &&
+                    event.isResolved &&
+                    event.card.canBeInPlay() && event.card.isInPlay() && !event.card.pendingDefeat &&
```

Comment:
What was the reason for adding these? Typically these checks are handle by the GameSystem in `immediateEffect`, there's a last-minute check that the target is still legal right before event resolution using the system's `canAffect()`. If it fails, then the event is cancelled. Then, in the triggered ability window, any triggered abilities for events that ended up getting cancelled are also cancelled. So this ability will never fire if the damage didn't actually end up happening, and we shouldn't need to explicitly filter that in the trigger.